src/scripts/main.go

```diff
@@ -50,85 +48,49 @@ func main() {
 		}
 	}
 
-	// Exit if the job status does not match the message send event and the message send event is not set to "always"
-	if !stringutils.IsEventMatchingStatus(conf.EventToSendMessage, conf.JobStatus) {
-		message := fmt.Sprintf("The job status %q does not match the status set to send alerts %q.", conf.JobStatus, conf.EventToSendMessage)
-		fmt.Println(message)
-		fmt.Println("Exiting without posting to Slack...")
-		os.Exit(0)
-	}
-
-	// Check if the branch and tag match their respective patterns and parse the invert match parameter
-	branchMatches, err := stringutils.IsPatternMatchingString(conf.BranchPattern, conf.JobBranch)
-	if err != nil {
-		log.Fatal("Error parsing the branch pattern:", err)
-	}
-	tagMatches, err := stringutils.IsPatternMatchingString(conf.TagPattern, conf.JobTag)
-	if err != nil {
-		log.Fatal("Error parsing the tag pattern:", err)
-	}
 	invertMatch, _ := strconv.ParseBool(conf.InvertMatchStr)
-	if !stringutils.IsPostConditionMet(branchMatches, tagMatches, invertMatch) {
-		fmt.Println("The post condition is not met. Neither the branch nor the tag matches the pattern or the match is inverted.")
-		fmt.Println("Exiting without posting to Slack...")
-		os.Exit(0)
+	ignoreErrors, _ := strconv.ParseBool(conf.IgnoreErrorsStr)
+	channels := strings.Split(conf.ChannelsStr, ",")
+
+	slackNotification := slack.Notification{
+		Status:                   conf.JobStatus,
+		Branch:                   conf.JobBranch,
+		Tag:                      conf.JobTag,
+		Event:                    conf.EventToSendMessage,
+		BranchPattern:            conf.BranchPattern,
+		TagPattern:               conf.TagPattern,
+		InvertMatch:              invertMatch,
+		InlineTemplate:           conf.InlineTemplate,
+		EnvVarContainingTemplate: conf.EnvVarContainingTemplate,
 	}
 
-	// Build the message body
-	template, err := jsonutils.DetermineTemplate(conf.InlineTemplate, conf.JobStatus, conf.EnvVarContainingTemplate)
+	modifiedJSON, err := slackNotification.BuildMessageBody()
 	if err != nil {
-		log.Fatalf("%v", err)
+		log.Fatalf("failed to build message body: %v", err)
+		os.Exit(0)
```

Comment:
~~ this `os.Exit(0) is superfluous here, as `log.Fatal` [prints the log message and quits with a code 1](https://pkg.go.dev/log#Fatalf). Exit code 0 is also used in the shell environment to indicate success, so if you were to keep an explicit exit here it should be a code greater than 0.

Reply:
👍